startupscript/post-startup.sh

```diff
@@ -202,6 +202,25 @@ if [[ "${LOG_IN}" == "true" ]]; then
     retry 5 "${SCRIPT_DIR}/git-setup.sh"
 fi
 
+##############################
+# passwordless setup
+##############################
+sudoers_file="/etc/sudoers"
+sudoers_d_file="/etc/sudoers.d/${USER_NAME}"
+
+# Check if there's an old rule in the main sudoers file that requires a password
+if grep -q "^${USER_NAME} ALL=(ALL:ALL) ALL" "${sudoers_file}"; then
+  echo "Found password-requiring rule for ${USER_NAME} in /etc/sudoers. Commenting it out."
+  
+  # Comment out the old rule in /etc/sudoers
+  sed -i "s/^${USER_NAME} ALL=(ALL:ALL) ALL/# ${USER_NAME} ALL=(ALL:ALL) ALL/" "${sudoers_file}"
+fi
+
+echo "${USER_NAME} ALL=(ALL) NOPASSWD:ALL" > "${sudoers_d_file}"
```

Comment:
IMO this should be done conditionally (maybe another parameter for this startup script command? or maybe more generally we can have this startup script check for/run all scripts in a custom startup scripts directory which the docker compose for vscode can mount)

There are apps like the AoU jupyter app that we may not want to give users sudo access

Reply:
i like having a custom directory